File: strings/14. Longest Common Prefix/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let prefix = "";
    
    shortestString = strs[0]
    for (let s of strs) {
        if (s.length < shortestString.length) {
            shortestString = s;
        }
    }
    
    for (let i = 0; i < shortestString.length; i++) {
        if (strs.every(currentString => currentString[i] === shortestString[i])) {
            prefix += shortestString[i];
        } else {
            break;
        }
    }
    
    return prefix;
};

File: strings/14. Longest Common Prefix/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = ""
        
        # Find the element with the smallest length
        # shortest_string = min(strs, key=len)
        shortest_string = strs[0]
        for s in strs:
            if len(s) < len(shortest_string):
                shortest_string = s
        
        # Iterate through the samelled string, and check each character
        for i in range(len(shortest_string)):
            # If all strings include the character, add it to the prefix
            if all(current_string[i] == shortest_string[i] for current_string in strs):
                prefix += shortest_string[i]
            else:
                break
        
        return prefix